File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp;
        for(int index = 0; index < nums.size(); ++index){
            if(mp.count(target - nums[index]))
                return {mp[target - nums[index]], index};
            mp[nums[index]] = index;
        }
        return {-1, -1};
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int index = 0; index < nums.length; ++index){
            int key = target - nums[index];
            if(map.containsKey(key)){
                int prevIndex = map.get(key);
                int[] result = {prevIndex, index};
                return result;
            }
            map.put(nums[index], index);
        }
        return null;
    }
}